content/en/docs/tasks/administer-cluster/migrating-from-dockershim/find-out-runtime-you-use.md

```diff
@@ -32,21 +32,69 @@ kubectl get nodes -o wide
 The output is similar to the following. The column `CONTAINER-RUNTIME` outputs
 the runtime and its version.
 
+For Docker Engine, the output is similar to this:
+
 ```none
-# For dockershim
 NAME         STATUS   VERSION    CONTAINER-RUNTIME
 node-1       Ready    v1.16.15   docker://19.3.1
 node-2       Ready    v1.16.15   docker://19.3.1
 node-3       Ready    v1.16.15   docker://19.3.1
 ```
+If your runtime shows as Docker Engine, you still might not be affected by the
+removal of dockershim in Kubernetes 1.24. [Check the runtime
+endpoint](#which-endpoint) to see if you use dockershim. If you don't use
+dockershim, you aren't affected. 
+
+For containerd, the output is similar to this:
 
 ```none
-# For containerd
 NAME         STATUS   VERSION   CONTAINER-RUNTIME
 node-1       Ready    v1.19.6   containerd://1.4.1
 node-2       Ready    v1.19.6   containerd://1.4.1
 node-3       Ready    v1.19.6   containerd://1.4.1
 ```
 
 Find out more information about container runtimes
-on [Container Runtimes](/docs/setup/production-environment/container-runtimes/) page.
+on [Container Runtimes](/docs/setup/production-environment/container-runtimes/)
+page.
+
+## Find out what container runtime endpoint you use {#which-endpoint}
+
+The container runtime talks to the kubelet over a Unix socket using the [CRI
+protocol](/docs/concepts/architecture/cri/), which is based on the gRPC
+framework. The kubelet acts as a client, and the runtime acts as the server.
+In some cases, you might find it useful to know which socket your nodes use. For
+example, with the removal of dockershim in Kubernetes 1.24 and later, Docker
+Engine cannot communicate with the kubelet if it listens on the
+`unix:///var/run/dockershim.sock` endpoint. 
```

Comment:
I don't understand this sentence. What communication Between Docker and kubelet is meant here? It is typically kubelet calls into the docker and it uses `docker.sock` to call to it. dokershim.sock is only exposed by kubelet to emulate the CRI from dockershim